fedimint-core/src/config.rs

```diff
@@ -269,10 +238,33 @@ impl<M> Default for ModuleGenRegistry<M> {
     }
 }
 
+/// Module **generation** (so passed to dkg, not to the module itself) config
+/// parameters
+#[derive(Debug, Clone, Default)]
+pub struct ConfigGenParams(serde_json::Value);
+
 pub type ServerModuleGenRegistry = ModuleGenRegistry<DynServerModuleGen>;
 
+impl ConfigGenParams {
+    pub fn to_typed<P: ModuleGenParams>(&self) -> anyhow::Result<P> {
+        serde_json::from_value(self.0.clone())
+            .map_err(|e| anyhow::Error::new(e).context("Invalid module params"))
+    }
+
+    pub fn from_typed<P: ModuleGenParams>(p: P) -> anyhow::Result<Self> {
+        Ok(Self(serde_json::to_value(p)?))
+    }
+}
+
 pub type CommonModuleGenRegistry = ModuleGenRegistry<DynCommonModuleGen>;
 
+/// Configs for each module's DKG
+///
+/// Note: in the future, we should make this one a
+/// ModuleRegistry<ConfigGenParams`, as each module **instance** will need a
+/// distinct config for dkg.
+pub type ConfigGenParamsRegistry = ModuleGenRegistry<ConfigGenParams>;
```

Comment:
I assume this would be somewhat involved rn, hence the note?

Reply:
Also, maybe the `\`` should be a `<`?